### PR TITLE
Fix: models_grouped always return opened data models

### PR DESCRIPTION
**Pull request by mcara**
`models_grouped` property of a `ModelContainer` should return file names instead of opened models when input are lists of files names under some circumstances. However, due to an incorrect indentation of a `if-block`, `models_grouped` *always* returns a list of opened (loaded in memory) data models.

This PR fixes the above.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -13,6 +13,12 @@ documentation
 
 - Fixed small typo in ``user_documentation`` docs. [#8178]
 
+datamodels
+----------
+
+- Fixed a bug in the ``ModelContiner`` due to which ``models_grouped``
```
Comment (hbushouse) [inline CHANGES.rst:19]:
```suggestion
- Fixed a bug in the ``ModelContainer`` data model, due to which the ``models_grouped``
```

**Review by hbushouse — CHANGES_REQUESTED** [review]

**Comment by hbushouse** [comment]
Need to run a full regression test on the PR branch.

**Comment by mcara** [comment]
Regression tests running here:
https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1200/

**Comment by hbushouse** [comment]
> Regression tests running here: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1200/

Regression test failures are benign and unrelated to this PR.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**